### ISSUE TITLE
Applying operators to undefined and null inputs seems inconsistent with excel

### ISSUE THREAD

**Issue by pbadenski** [formulajs/formulajs#48]
Below for '+' operator:

Excel | formulajs
----- | ---------
empty_cell + empty_cell = 0 | ADD(undefined, undefined) = Error: #VALUE!
empty_cell + 1 = 1 | ADD(undefined, 1) = Error: #VALUE!
1 + empty_cell = 1 | ADD(1, undefined) = Error: #VALUE!
? (see 1. below) | null + null = NaN
? (see 1. below) | null + 1 = NaN
? (see 1. below) | 1 + null = NaN

1. I'm not sure what would be seen as equivalent of null in Excel. In formulajs `SUM(null)` returns 0, so that would imply at the moment null is seen as an equivalent of an empty cell. There could be an argument for treating null as `NA` - which I personally would think is semantically more accurate - this is a separate conversation. As things are now - probably matching the behaviour of SUM would be a good first step.

From what I've seen this is an issue for at least: ADD, MINUS, MULTIPLY and DIVIDE. I can provide more detailed breakdown or even pull request if changing it feels like the right direction.

Related to #47

**Comment by nicolashefti** [comment]
#N/A is an [Error](https://support.microsoft.com/en-us/office/na-function-5469c2d1-a90c-4fb5-9bbc-64bd9bb6b47c?ui=en-US&rs=en-US&ad=US) whereas null intentional [absence of any object value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null). There is nothing « error-ish » in null.

**Comment by pbadenski** [comment]
True, but, if you consider the description of both - there seems to be a similarity as to intention:

Excel:
`N/A is the error value that means "no value is available." Use NA to mark empty cells.`

JavaScript:
`The value null represents the intentional absence of any object value.`

I agree - in Excel NA() is represented as an error - but I think that speaks more to the fact  they chose a crude way to represent the underlying concept.

Regardless of where we get on the topic of null vs NA - the first part of the issue still should be resolved I think - should I suggest proposal all in a single pull request, combined with #47?

**Comment by pbadenski** [comment]
@nicolashefti Any thoughts on this?

**Comment by nicolashefti** [comment]
Sorry I missed your comment on the issue.

What is the current status of the initial issue? Thanks to you, we are now more and more consistent with Excel, returning the Formula.js error.na instead of undefined/0. I will review #55 this weekend latest.

There was still the question about JS "null" being the equivalent of Excel "NA". Does this still have an impact?

**Comment by pbadenski** [comment]
From what I see there is still 2 pull requests to go - #55 and another one for "lib/text" I haven't submitted yet.

It matters for the use case we have - we're allowing users to leverage formula for manipulating financial time series. We return "null" values for holidays and weekends, and in consequence expect "NA()" like behaviour. Eg: on a weekend when all values are "null" we expect SUM to be "null" - not "0" as it would at the moment.

**Comment by nicolashefti** [comment]
I am still a bit skeptical about treating `null` as an error. For me, even if it is intentional, `null` is closer to the JS `undefined` as `new Error()`.

Why don't you return error.na (using #59) within your "financial time series", before the sum? You would get the expected result with the current implementation?

An example of your implementation might help here.

**Comment by pbadenski** [comment]
True - we can replace `null` with `error.na`, so that pretty much covers the use case :)

I see now where you're coming from. It might be confusing for JS developers if null is behaving radically different than undefined - especially when resulting in `error.na` which under the hood is represented as `Error`.

Let's wrap it up here for now I think - I think I have a good solution here, and sounds like this change is definitely controversial :)

**Comment by nicolashefti** [comment]
I think we can close this one.
